lib/streamlit/elements/streamlit_plotly_theme.py

```diff
@@ -0,0 +1,193 @@
+CATEGORY_8 = "#000009"
+CATEGORY_9 = "#000010"
+
+SEQUENTIAL_10 = "#000011"
+SEQUENTIAL_11 = "#000012"
+SEQUENTIAL_12 = "#000013"
+SEQUENTIAL_13 = "#000014"
+SEQUENTIAL_14 = "#000015"
+SEQUENTIAL_15 = "#000016"
+SEQUENTIAL_16 = "#000017"
+SEQUENTIAL_17 = "#000018"
+SEQUENTIAL_18 = "#000019"
+SEQUENTIAL_19 = "#000020"
+
+DIVERGING_20 = "#000021"
+DIVERGING_21 = "#000022"
+DIVERGING_22 = "#000023"
+DIVERGING_23 = "#000024"
+DIVERGING_24 = "#000025"
+DIVERGING_25 = "#000026"
+DIVERGING_26 = "#000027"
+DIVERGING_27 = "#000028"
+DIVERGING_28 = "#000029"
+DIVERGING_29 = "#000030"
+DIVERGING_30 = "#000031"
+
+INCREASING_31 = "#000032"
+DECREASING_32 = "#000033"
+TOTAL_33 = "#000034"
+
+GRAY_30_34 = "#000035"
+GRAY_70_35 = "#000036"
+GRAY_90_36 = "#000037"
+BG_COLOR_37 = "#000038"
+FADED_TEXT_05_38 = "#000039"
+BG_MIX_39 = "#000040"
+
+BODY_FONT = '"Source Sans Pro", sans-serif'
+
+
+# Plotly represents continuous colorscale through an array of pairs.
+# The pair's first index is the starting point and the next pair's first index is the end point.
+# The pair's second index is the starting color and the next pair's second index is the end color.
+# For more information, please refer to https://plotly.com/python/colorscales/
+
+streamlit_colorscale = [
+    [0.0, SEQUENTIAL_10],
+    [0.1111111111111111, SEQUENTIAL_11],
+    [0.2222222222222222, SEQUENTIAL_12],
+    [0.3333333333333333, SEQUENTIAL_13],
+    [0.4444444444444444, SEQUENTIAL_14],
+    [0.5555555555555556, SEQUENTIAL_15],
+    [0.6666666666666666, SEQUENTIAL_16],
+    [0.7777777777777778, SEQUENTIAL_17],
+    [0.8888888888888888, SEQUENTIAL_18],
+    [1.0, SEQUENTIAL_19],
+]
+
+pio.templates["streamlit"] = go.layout.Template(
+    data=go.layout.template.Data(
+        candlestick=[
+            go.layout.template.data.Candlestick(
+                decreasing=go.candlestick.Decreasing(
+                    line=go.candlestick.decreasing.Line(color=DECREASING_32)
+                ),
+                increasing=go.candlestick.Increasing(
+                    line=go.candlestick.increasing.Line(color=INCREASING_31)
+                ),
+            )
+        ],
+        contour=[go.layout.template.data.Contour(colorscale=streamlit_colorscale)],
+        contourcarpet=[
+            go.layout.template.data.Contourcarpet(colorscale=streamlit_colorscale)
+        ],
+        heatmap=[go.layout.template.data.Heatmap(colorscale=streamlit_colorscale)],
+        histogram2d=[
+            go.layout.template.data.Histogram2d(colorscale=streamlit_colorscale)
+        ],
+        icicle=[
+            go.layout.template.data.Icicle(textfont=go.icicle.Textfont(color="white"))
+        ],
+        sankey=[
+            go.layout.template.data.Sankey(
+                textfont=go.sankey.Textfont(color=GRAY_70_35, family=BODY_FONT)
+            )
+        ],
+        scatter=[
+            go.layout.template.data.Scatter(
+                marker=go.scatter.Marker(line=go.scatter.marker.Line(width=0))
```

Comment:
nit: since `width=0` is not color related, this needs to be defined here? Just want to make sure we are only adding the minimal aspects in our placeholder theme.

Reply:
There was a p1 or p2 issue that Andreas created where he wanted to delete the markers. Developers can still override this if they want so not a big deal imo.